docs: pwa (#788)
front-commerce/developers.front-commerce.com: docs/02-guides/pwa-setup.mdx

---
title: PWA Setup
description: This guide explains how to setup your PWA.
sidebar_position: 5
---

import Figure from "@site/src/components/Figure";

<p>{frontMatter.description}</p>

## What is a PWA

> PWA stands for Progressive Web App, it will allow your shop to be installed as
> an application on end user devices (mobile, desktop, tablets).
> 
> It allow advanced caching on device, fastest navigation, offline support and
> much more.

Reviewer: ```suggestion
[PWA stands for Progressive Web App](https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps). This technology allows the user to install your store as a native application on their device. It also brings advanced caching capabilities, offline support and much more.
```

Author: fixed with your suggestion

Front-Commerce integrates PWA capabilities and is available with a few
configuration.

## Setup PWA

### `webmanifest` configuration

> The [`webmanifest`](https://developer.mozilla.org/en-US/docs/Web/Manifest) will
> define meta informations about your store. To setup this informations, you needs
> to add the following configuration in your `front-commerce.config.ts` file:

Reviewer: ```suggestion
The [`webmanifest`](https://developer.mozilla.org/en-US/docs/Web/Manifest)
defines meta information about your store. To setup this webmanifest, you need
to add the following configuration in your `front-commerce.config.ts` file:
```

Author: fixed with your suggestion

```typescript title="front-commerce.config.ts"
import { defineConfig } from "@front-commerce/core/config";
import themeChocolatine from "@front-commerce/theme-chocolatine";
import magento2 from "@front-commerce/magento2";
import storesConfig from "./app/config/stores";
import cacheConfig from "./app/config/caching";

export default defineConfig({
  extensions: [magento2(storesConfig), themeChocolatine()],
  stores: storesConfig,
  cache: cacheConfig,
  configuration: {
    providers: [],
  },
  // highlight-start
  pwa: {
    appName: "ACME Store",
    shortName: "ACME Store",
    themeColor: "#fbb03b",
    icon: "public/pwa-icon.png",
    maskableIcon: "public/pwa-masked-icon",
    ttl: 60,
  },
  // highlight-end
});
```

> Here's a description of options that can be configured:

Reviewer: ```suggestion
Here is a list of the available options:
```

Author: fixed with your suggestion

| -------------- | -------- | -------------------------------------------------------------------------------------------------------------------- |
| `appName`      | `string` | The name of your PWA                                                                                                 |
| `shortName`    | `string` | The short name of your PWA, it will be displayed on the home screen of your customer                                 |
| `themeColor`   | `string` | The theme color of your PWA, it will define theme for your App, we recommend you to use your website primary color   |
| `icon`         | `string` | The icon of your PWA, it will be displayed on the home screen of your end user device, it must be at least 512x512   |
| `maskableIcon` | `string` | The maskable icon of your PWA, it will be displayed on the home screen of your customer, it must be at lease 512x512 |
| `ttl`          | `number` | The time to live of your PWA Cache in milliseconds, the default value is of 7 days                                   |

### Icons setup

> PWA require multiple icons to be defined in your project that will be used by
> device to identify your PWA on end-user device screen.

Reviewer: ```suggestion
A PWA requires multiple icons to be defined in your project that will be used by
the device to identify the application on the end-user device screen.
```

Author: fixed with your suggestion

Two types of icons are needed:

#### `icon`

This icon is a standard icon that will be displayed over a background color
depending on the end-user device configuration and OS.

<Figure caption="Example of standard icons displayed on Android">

![Example on standard icons displayed on android](./assets/standard_icons.png)

</Figure>

> Source: [web.dev](https://web.dev/maskable-icon/)

#### `maskableIcon`

This icon must be a squared icon and will be cropped in a shape defined by the
device

<Figure caption="Example of maskable icons displayed on Android">

![Example of maskable icons displayed on android](./assets/maskable_icons.png)

</Figure>

> Source: [web.dev](https://web.dev/maskable-icon/)

You can read more about icons here :
[https://web.dev/maskable-icon/](https://web.dev/maskable-icon/)

## Install PWA

Once everything is setup, you will be able to install your website as an
application.

:::caution

Except when on localhost, PWA installation is only possible with **HTTPS**. If
you're using Front-Commerce Cloud, HTTPS is enabled by default. If you use

> you're own server, you need to enable HTTPS.

Reviewer: not sure we need to add that in 2023 :)

Author: removed

:::

> Depending on the browser you're using, the browser may display a prompt to
> invite you to install the application. Otherwise, please check your browser
> documentation on how to install PWA.

Reviewer: ```suggestion
Depending on the browser, a prompt to
invite the customer to install the application is different:
```

Author: fixed with your suggestion

- Chrome: https://support.google.com/chrome/answer/9658361

> - Safari : https://web.dev/learn/pwa/installation/#ios-and-ipados-installation
> - Firefox :
>   https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Guides/Installing
> - Edge :
>   https://learn.microsoft.com/en-us/microsoft-edge/progressive-web-apps-chromium/ux

Reviewer: ```suggestion
- Safari: https://web.dev/learn/pwa/installation/#ios-and-ipados-installation
- Firefox:
  https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Guides/Installing
- Edge:
  https://learn.microsoft.com/en-us/microsoft-edge/progressive-web-apps-chromium/ux
```

Author: fixed with your suggestion
